### ISSUE TITLE
No option to allow biometrics for a database

### ISSUE THREAD

**Issue by TheDsm112** [Kunzisoft/KeePassDX#900]
When entering a database and tapping the fingerprint button, it says "type in the password, then click the advanced unlock button."
When going into the advanced unlock settings, there is a toggle for biometric unlocking. Even when enabled, the app does not allow the use of a fingerprint.

**Comment by J-Jamet** [comment]
You did not correctly follow the wiki for linking your database with biometric recognition.
https://github.com/Kunzisoft/KeePassDX/wiki/Advanced-Unlocking

`type in the password, then click the advanced unlock button`: this sentence is clear, you must first write your password and then press the biometric button.

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
> When entering a database and tapping the fingerprint button, it says "type in the password, then click the advanced unlock button."
> When going into the advanced unlock settings, there is a toggle for biometric unlocking. Even when enabled, the app does not allow the use of a fingerprint.

Same thing that happened to me

**Comment by J-Jamet** [comment]
What do you mean?

**Comment by J-Jamet** [comment]
Please understand that without context, I cannot solve your problems.
You did not use the bug opening pattern that allows to describe a problem precisely.
You don't indicate any information, neither the steps that precisely describe your actions, nor the hardware and software versions you use.
So for me, you didn't read the documentation and you simply misuse the application, you need a minimum of involvement that just saying "it doesn't work".

**Comment by TheDsm112** [comment]
> You did not correctly follow the wiki for linking your database with biometric recognition.
> https://github.com/Kunzisoft/KeePassDX/wiki/Advanced-Unlocking
> 
> `type in the password, then click the advanced unlock button`: this sentence is clear, you must first write your password and then press the biometric button.

The instructions, to me, indicated that the user would have to go into the settings of the database to enable biometrics, given it says "click the advanced unlock button." Rather, it should say "Type in the password, then tap the fingerprint button" or add an option to do so directly in the database settings.

**Comment by J-Jamet** [comment]
If you choose "Device credential unlocking", it is not a biometric button but a lightning button, "advanced unlock" button is a suitable sentence for both cases.

>  or add an option to do so directly in the database settings.

I don't understand what you mean, which option do you want to put in the settings? Clicking a button in the settings would be much more painful than placing it above the master password field.

**Comment by hefty-fading-culprit-epiphany-drew** [comment]
Omega lul @J-Jamet just figured out I have to write password and click on fingerprint option instead of entering password and opening database and going to advance unlock to configure biometrics.
Currently I don't think that it is phrased the right way, it should be instead 
"Enter password and click on this fingerprint icon to register fingerprint unlocking"

**Comment by TheDsm112** [comment]
I followed your instructions @J-Jamet and now it works. Thank you.

**Comment by robertour** [comment]
TLDR; RTFI: https://github.com/Kunzisoft/KeePassDX/wiki/Advanced-Unlocking

-----

@J-Jamet, thanks for pointing to the instructions. Following them carefully fixes the "issue", but I made exactly the same mistake as @TheDsm112. I don't think this is clear in the interface. As you said, it says:

`type in the password, then click the advanced unlock button`

And I was looking for such a button, which I found on the `"top corner (3 vertical dots)" -> Settings -> Advancing unlocking`. Ok, now I see it is not the same. I mean, `Advancing unlocking` is not the same as `advanced unlock button`, but it is pretty close.

So, I agree that there is a bit of an issue with the message, so maybe you can just change that to make it clear that it is the orange fingerprint icon (https://github.com/Kunzisoft/KeePassDX/blob/60ba058515144b247018df7bcf55b53d0d6b0efd/app/src/main/res/values/strings.xml#L297)

Other than that, it works perfectly.

**Comment by XOCODE-OP** [comment]
On my stock android this works, on my calyx when I tap the fingerprint button it flashes but nothing pops up.
Keepass2android works fine on calyx.
Some permission missing or something?

I made logs!
![Screenshot_20220930-154442](https://user-images.githubusercontent.com/12516312/193208044-9798fd92-58da-4a95-84c4-63a3ff8755be.jpg)
![Screenshot_20220930-154434](https://user-images.githubusercontent.com/12516312/193208085-88c4eae4-9c59-4278-b51e-10770f71be63.jpg)

**Comment by J-Jamet** [comment]
@XOCODE-OP look at #1380
